src/wmtk/io/Cache.cpp

```diff
@@ -90,48 +126,48 @@ Cache::~Cache()
     }
 }
 
-const std::filesystem::path& Cache::create_unique_file(
+const std::filesystem::path& Cache::create_cache_file(
     const std::string& filename,
-    const std::string& extension,
-    size_t max_tries)
+    const std::string& extension)
 {
-    const std::string timestamp = number_to_hex(nanoseconds_timestamp());
+    throw_memory_only_mode();
```

Comment:
is this worth a throw or just a warning?

Reply:
either way i prefer the 2 lines of "if memory only mode throw memory only mode" - this code doesn't really imply a branch and the change in control flow isn't obvious